src/pip/_internal/index.py

```diff
@@ -874,10 +874,15 @@ def find_requirement(self, req, upgrade):
         Returns a Link if found,
         Raises DistributionNotFound or BestVersionAlreadyInstalled otherwise
         """
-        hashes = req.hashes(trust_internet=False)
-        best_candidate_result = self.find_best_candidate(
-            req.name, specifier=req.specifier, hashes=hashes,
-        )
+        if req.is_pinned and req.satisfied_by is not None:
```

Comment:
We'd want to handle this case in `Resolver._check_skip_installed` instead of here.